pkg/provider/store/object_store.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"context"
+
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/go-kit/log"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/objstore/client"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type Store struct {
+	TsdbPath     string
+	ObjectKey    string
+	ObjectConfig string
+}
+
+var S Store
```

Comment:
This is not a good practice. Why do you want to have a package-level variable? Why don't we have a `NewStore` function to create a new store? We can get the bucket as a dependency and associate the function below to the store struct type.